src/run-function.ts

```diff
@@ -19,32 +23,59 @@ export const RunFunction = async (
     slackApiUrl: env["SLACK_API_URL"],
   });
 
-  // We don't catch any errors the handlers may throw, we let them throw, and stop the process
-  const {
-    completed = true,
-    outputs = {},
-    error,
-  } = await functionModule.default({
+  const returnArgs: FunctionHandlerReturnArgs = {
+    completed: false,
+    outputs: {},
+    error: undefined,
+  };
+
+  const functionContext: FunctionContext = {
     inputs,
     env,
     token,
-    // pass along the full event for convenience/future properties
     event: body.event,
-  });
+  };
+
+  // We don't catch any errors the handlers may throw, we let them throw, and stop the process
+  if (functionModule.default.constructor.name === "AsyncFunction") {
+    const functionToRun = functionModule.default as AsyncFunctionHandler;
+
+    const {
+      completed = true,
+      outputs = {},
+      error,
+    } = await functionToRun(functionContext);
+
+    returnArgs.completed = completed;
```

Comment:
I couldn't think of a simpler way to diverge between sync/async, and consolidate the return values, so open to any better ideas here 😄